stdlib/public/core/Sequence.swift

```diff
@@ -1206,24 +1189,19 @@ extension Sequence where
     _precondition(n >= 0, "Can't drop a negative number of elements from a sequence")
     if n == 0 { return AnySequence(self) }
 
-    // FIXME: <rdar://problem/21885650> Create reusable RingBuffer<T>
     // Put incoming elements from this sequence in a holding tank, a ring buffer
     // of size <= n. If more elements keep coming in, pull them out of the
     // holding tank into the result, an `Array`. This saves
     // `n` * sizeof(Iterator.Element) of memory, because slices keep the entire
     // memory of an `Array` alive.
     var result: [Iterator.Element] = []
-    var ringBuffer: [Iterator.Element] = []
-    var i = ringBuffer.startIndex
-
+    var ringBuffer = RingBuffer<Iterator.Element>(capacity: n)
+    
     for element in self {
-      if ringBuffer.count < n {
-        ringBuffer.append(element)
-      } else {
-        result.append(ringBuffer[i])
-        ringBuffer[i] = element
-        i = ringBuffer.index(after: i) % n
+      if ringBuffer.isFull, let first = ringBuffer.first {
```

Comment:
The `ringBuffer.append(element)` later will be equivalent to `popFirst()`.

I expect that popFirst is less performant on the current ring buffer implementation. It could use a `startIndex` and an `endIndex` to change that.